### PR TITLE
Feature/add new inference meta

### PR DESCRIPTION
**Pull request by rrcarlosrodriguez**

### CODE REVIEW

File: gst-libs/gst/r2inference/gstinferenceprediction.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+};
+
+/**
+ * Abstraction that represents a prediction
+ */
+struct _GstInferencePrediction
+{
+  GstMiniObject base;
+
+  guint64 id;
```
Comment (migueltaylor) [inline gst-libs/gst/r2inference/gstinferenceprediction.h]: Maybe change this variable name to something like `prediction_id` to avoid confusion with other ids that may be added in the future (like `model_id` or `object_id` for a tracker)

File: gst-libs/gst/r2inference/gstinferenceclassification.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+static void classification_reset (GstInferenceClassification * self);
+
+static gdouble *probabilities_copy (const gdouble * from, gint num_classes);
+
+static void
+classification_reset (GstInferenceClassification * self)
+{
+  g_return_if_fail (self);
+
+  self->class_id = -1;
```
Comment (migueltaylor) [inline gst-libs/gst/r2inference/gstinferenceclassification.c]: Define `DEFAULT_CLASS_ID`,  `DEFAULT_CLASS_PROB`, `DEFAULT_NUM_CLASSES`, etc.  To avoid hard-coded values.

File: gst-libs/gst/r2inference/gstinferencepostprocess.c [hunk 3]
```diff
@@ -251,6 +248,67 @@ gst_create_boxes (GstVideoInference * vi, const gpointer prediction,
...
+  gint num_classes = 0;
+  const gchar *label = NULL;
+  gchar **labels = NULL;
+
+  g_return_val_if_fail (vi != NULL, NULL);
+
+  num_classes = predsize / sizeof (gfloat);
+
+  /* FIXME: This is just dumb */
+  if (sizeof (gfloat) != sizeof (gdouble)) {
```
Comment (migueltaylor) [inline gst-libs/gst/r2inference/gstinferencepostprocess.c]: Is this condition necessary?

File: gst-libs/gst/r2inference/gstvideoinference.c [hunk 4]
```diff
@@ -48,16 +49,21 @@ GST_DEBUG_CATEGORY_STATIC (gst_video_inference_debug_category);
 enum
 {
   NEW_PREDICTION_SIGNAL,
+  NEW_INFERENCE_SIGNAL,
```
Comment (migueltaylor) [inline gst-libs/gst/r2inference/gstvideoinference.c]: Why did you add this signal? What is the difference between `NEW_PREDICTION_SIGNAL` and `NEW_INFERENCE_SIGNAL`? If there is a reason to have both, it should be documented in the code.

File: gst-libs/gst/r2inference/gstinferenceprediction.c [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+node_copy (gconstpointer node, gpointer data)
+{
+  GstInferencePrediction *self = (GstInferencePrediction *) node;
+
+  return prediction_copy (self);
+}
+
+static gboolean
+node_assign (GNode * node, gpointer data)
+{
```
Comment (migueltaylor) [inline gst-libs/gst/r2inference/gstinferenceprediction.c]: pointer guard `Node`

File: gst-libs/gst/r2inference/gstinferenceprediction.c [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+      g_list_copy_deep (self->classifications, (GCopyFunc) classification_copy,
+      NULL);
+
+  return other;
+}
+
+static gpointer
+node_copy (gconstpointer node, gpointer data)
+{
+  GstInferencePrediction *self = (GstInferencePrediction *) node;
```
Comment (migueltaylor) [inline gst-libs/gst/r2inference/gstinferenceprediction.c]: pointer guard `Node`

File: ext/r2inference/gsttinyyolov2.c [hunk 7]
```diff
@@ -280,16 +288,33 @@ gst_tinyyolov2_preprocess (GstVideoInference * vi,
 
 static gboolean
 gst_tinyyolov2_postprocess (GstVideoInference * vi, const gpointer prediction,
-    gsize predsize, GstMeta * meta_model, GstVideoInfo * info_model,
+    gsize predsize, GstMeta * meta_model[2], GstVideoInfo * info_model,
     gboolean * valid_prediction)
+{
+  gboolean ret = TRUE;
+
```
Comment (rrcarlosrodriguez) [inline ext/r2inference/gsttinyyolov2.c]: Pointer guards for info_model, meta_model

File: gst-libs/gst/r2inference/gstvideoinference.c (hunk 4 above)
Comment (rrcarlosrodriguez) [reply gst-libs/gst/r2inference/gstvideoinference.c]: This is to avoid breaking the old approach! Once the whole new approach is ready the old signal will be removed
